### PR TITLE
Username checking in native auth requests

### PR DESCRIPTION
**Pull request by Yuki-YuXin**
Put back ArgUtils.validateNonNullArg(username, "username") in all initial native auth requests.

Company PRs:
msal: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/2080
native sample app: https://github.com/Azure-Samples/ms-identity-ciam-native-auth-android-sample/pull/24

### CODE REVIEW

**Review by SaurabhMSFT — APPROVED** [review]

File: common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/requests/resetpassword/ResetPasswordStartRequest.kt [hunk 1]
```diff
@@ -53,6 +53,7 @@ data class ResetPasswordStartRequest private constructor(
             headers: Map<String, String?>
         ): ResetPasswordStartRequest {
             // Check for empty Strings and empty Maps
+            ArgUtils.validateNonNullArg(username, "username")
```
Comment (SammyO) [inline common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/requests/resetpassword/ResetPasswordStartRequest.kt:56]: why was this removed in the first place?

File: common4j/src/test/com/microsoft/identity/common/java/nativeauth/providers/NativeAuthRequestHandlerTest.kt [hunk 2]
```diff
@@ -96,8 +96,8 @@ class NativeAuthRequestHandlerTest {
         )
 
     // signup start tests
-    @Test
-    fun testSignUpStartWithEmptyUsernameShouldNotThrowException() {
+    @Test(expected = ClientException::class)
+    fun testSignUpStartWithEmptyUsernameShouldThrowException() {
```
Comment (SammyO) [inline common4j/src/test/com/microsoft/identity/common/java/nativeauth/providers/NativeAuthRequestHandlerTest.kt:100]: Wasn't this changed (i.e. the exception removed) as part of last year's work to align some of our errors and return an InvalidUsernameError in the case of an empty string? Giannis worked on that I believe.

File: common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/requests/resetpassword/ResetPasswordStartRequest.kt (hunk 1 above)
Comment (Yuki-YuXin) [reply common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/requests/resetpassword/ResetPasswordStartRequest.kt:56]: Not sure about the reason but I would prefer to keep the checking here.

File: common4j/src/test/com/microsoft/identity/common/java/nativeauth/providers/NativeAuthRequestHandlerTest.kt (hunk 2 above)
Comment (Yuki-YuXin) [reply common4j/src/test/com/microsoft/identity/common/java/nativeauth/providers/NativeAuthRequestHandlerTest.kt:100]: Yes. I think so. But either `CommandResultUtil.kt`
```kotlin
if (this.status != ICommandResult.ResultStatus.COMPLETED) {
        var exception: Exception? = null
        var exceptionMessage: String? = ""

        if (this.result is Exception) {
            exception = this.result as Exception
            exceptionMessage = exception.message
        }

        return com.microsoft.identity.common.java.nativeauth.controllers.results.INativeAuthCommandResult.UnknownError(
            error = UNSUCCESSFUL_COMMAND_ERROR,
            errorDescription = exceptionMessage,
            exception = exception,
            correlationId = this.correlationId
        ) as ExpectedType
```

or` NativeAuthPublicClientApplication.kt`
```kotlin
return withContext(Dispatchers.IO) {
            try {
                verifyNoUserIsSignedIn()

                if (username.isBlank()) {
                    return@withContext SignUpError(
                        errorType = ErrorTypes.INVALID_USERNAME,
                        errorMessage = "Empty or blank username",
                        correlationId = "UNSET"
                    )
                }
```

can handle empty username. 

I think the latter one act the main role of returning InvalidUsernameError before sending out the request/validating request parameters. Thus, I would prefer to keep the ThrowException here.

Comment (SammyO) [reply common4j/src/test/com/microsoft/identity/common/java/nativeauth/providers/NativeAuthRequestHandlerTest.kt:100]: Some thoughts: 
- if we're already doing a username is empty check in the interface, we don't need to do it again in the API layer.
- we shouldn't just be throwing exceptions. An exception is a pretty heavy mechanism to give feedback to a developer. We should aim for something more graceful than that.
- the fact that this "throwing an exception" was removed from this API layer a few months ago, means we had a reason why. If that reason has changed, let's talk about it. We can just revert the work we did a few months ago without good reason.

Comment (Yuki-YuXin) [reply common4j/src/test/com/microsoft/identity/common/java/nativeauth/providers/NativeAuthRequestHandlerTest.kt:100]: Will abandon this PR.
